### PR TITLE
Prevent clients from accessing LAN subnets

### PR DESCRIPTION
**Pull request by soffokl**
Another part of this feature is here: https://github.com/mysteriumnetwork/node/pull/1420

This PR enables internal OpenVPN packet filtering, which is platform-independent.

### CODE REVIEW

File: openvpn/middlewares/server/auth/middleware.go [hunk 1]
```diff
@@ -24,6 +24,16 @@ import (
 	"github.com/mysteriumnetwork/go-openvpn/openvpn/management"
 )
 
+const filterLANTemplate = `client-pf %d
+ [CLIENTS DROP]
+ [SUBNETS ACCEPT]
+ -10.0.0.0/8
+ -172.16.0.0/12
+ -192.168.0.0/16
```
Comment (zolia) [inline openvpn/middlewares/server/auth/middleware.go]: Protected subnets should be passed from the node. We have a parameter for those.

Comment (soffokl) [reply openvpn/middlewares/server/auth/middleware.go]: Changed it to pass subnets from outside.

File: openvpn/middlewares/server/auth/middleware.go [hunk 2]
```diff
@@ -176,8 +191,13 @@ func (m *middleware) authenticateClient(clientID, clientKey int, username, passw
 	}
 
 	if authenticated {
+		if err := filterSubnets(m.commandWriter, clientID, m.filter); err != nil {
```
Comment (Waldz) [inline openvpn/middlewares/server/auth/middleware.go]: Lets not couple firewall configuration to authorisation hook.
Also would be good to have it in `node` repo as it is for Linux firewall rules.

Comment (soffokl) [reply openvpn/middlewares/server/auth/middleware.go]: It's not a Linux firewall configuration. 
It's OpenVPN packet filtering which is applied during authorization.

Comment (soffokl) [reply openvpn/middlewares/server/auth/middleware.go]: Extracted it into dedicated middleware, it seems like it works this way.

**Pull request merged**